lib/smart_proxy_dns_powerdns/backend/mysql.rb

```diff
@@ -35,15 +35,20 @@ def create_record domain_id, name, type, content
       name = connection.escape(name)
       content = connection.escape(content)
       type = connection.escape(type)
-      connection.query("INSERT INTO records (domain_id, name, ttl, content, type) VALUES (#{domain_id}, '#{name}', #{ttl.to_i}, '#{content}', '#{type}')")
+      connection.query("INSERT INTO records (domain_id, name, ttl, content, type, change_date) VALUES (#{domain_id}, '#{name}', #{ttl.to_i}, '#{content}', '#{type}', UNIX_TIMESTAMP())")
       connection.affected_rows == 1
     end
 
     def delete_record domain_id, name, type
       name = connection.escape(name)
       type = connection.escape(type)
       connection.query("DELETE FROM records WHERE domain_id=#{domain_id} AND name='#{name}' AND type='#{type}'")
-      connection.affected_rows >= 1
+      ret = connection.affected_rows >= 1
+      if ret
+        connection.query("UPDATE records SET change_date=UNIX_TIMESTAMP() WHERE domain_id=#{domain_id} AND type='SOA'")
+        ret = connection.affected_rows == 1
```

Comment:
Another question I have is: do we need to fail hard when update of the SOA fails?
My guess would be yes because usually, if the SOA serial does not get bumped, no other DNS slaves get notified. Also, forwarders would no catch up on the changes until the record TTL has expired (since no change in SOA serial means no change to the zone content).

Maybe raising a proper exception with plenty of information would be the best way to go.
All of this would need to be run inside a transaction though because this could lead to the following bad scenario otherwise:

* delete record → success
* update SOA → fails, raise an exception
* user tries again
* delete record →no record deleted → does not update SOA :(